### PR TITLE
CVA6: Questasim support

### PR DESCRIPTION
**Pull request by MarioOpenHWGroup**
This PR has the commits of PR #1603 but  breaks the branch naming convention of `cvvdev/` to make the CI work.

### CODE REVIEW

File: .gitlab-ci/cva6.yml [hunk 1]
```diff
@@ -182,7 +182,7 @@ pub_smoke:
     # The list of files must NOT fail on various DV_SIMULATORS values, so use 'v*_sim' to match
     # 'veri-testharness_sim', 'vcs-testharness_sim' and 'vcs-uvm_sim' (one of them always applies,
     # at least until new RTL simulator configurations are added.)
-    - for i in cva6/sim/*/v*_sim/*.log.iss ; do head -10000 $i > artifacts/logs/$(basename $i).head ; done
+    - for i in cva6/sim/*/*_sim/*.log.iss ; do head -10000 $i > artifacts/logs/$(basename $i).head ; done
```
Comment (cathales) [inline .gitlab-ci/cva6.yml:185]: If we are adding questa to the CI, I think the comment above should be updated accordingly.

File: .gitlab-ci/cva6.yml [hunk 2]
```diff
@@ -167,7 +167,7 @@ pub_smoke:
       artifacts: true
   parallel:
     matrix:
-      - DV_SIMULATORS: ["veri-testharness,spike","vcs-testharness,spike","vcs-uvm,spike" ]
+      - DV_SIMULATORS: ["veri-testharness,spike","vcs-testharness,spike","vcs-uvm,spike","questa-testharness,spike" ]
```
[the remark this answers is not in the document]
Comment (MarioOpenHWGroup) [reply .gitlab-ci/cva6.yml:170]: If it fits I will remove the CI job for questa to merge the changes in cva6/dev and later I will open another PR with the CI changes.

File: .gitlab-ci/cva6.yml (hunk 1 above)
Comment (MarioOpenHWGroup) [reply .gitlab-ci/cva6.yml:185]: Totally agree, mb

File: .gitlab-ci/cva6.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cathales) [reply .gitlab-ci/cva6.yml:170]: I aggree, this way we can accept your changes and fix the CI environment later

**Comment by MarioOpenHWGroup** [comment]
@zchamski @cathales trigger the pipeline if possible :smile:

**Pull request merged**